unittest/server/src/ivi_controller_uinttests.cpp

```diff
@@ -0,0 +1,3475 @@
+ *                      -# Mocking the weston_plugin_api_get() does return an object
+ *                      -# Calling the wet_module_init()
+ *                      -# Verification point:
+ *                         +# wet_module_init() must not return 0
+ *                         +# weston_plugin_api_get() must be called once time
+ *                         +# wl_global_create() must be called once time
+ *                         +# weston_load_module() not be called
+ */
+TEST_F(ControllerTests, wet_module_init_cannotCreateGlobalWmIviInterface)
+{
+    struct ivi_layout_interface *lp_iviLayoutInterface = &g_iviLayoutInterfaceFake;
+    SET_RETURN_SEQ(weston_plugin_api_get, (const void**)&lp_iviLayoutInterface, 1);
+
+    ASSERT_NE(wet_module_init(&m_westonCompositor, nullptr, nullptr), 0);
+
+    ASSERT_EQ(weston_plugin_api_get_fake.call_count, 1);
+    ASSERT_EQ(wl_global_create_fake.call_count, 1);
+    ASSERT_EQ(weston_load_module_fake.call_count, 0);
+}
+
+/** ================================================================================================
+ * @test_id             wet_module_init_cannotInitIviInputModule
+ * @brief               Test case of wet_module_init() where weston_plugin_api_get() success, return an object
+ *                      and wl_global_create() success, return an object but ms_inputControllerModuleValue is -1
+ * @test_procedure Steps:
+ *                      -# Set ms_inputControllerModuleValue is -1
+ *                      -# Mocking the weston_plugin_api_get() does return an object
+ *                      -# Mocking the wl_global_create() does return an object
+ *                      -# Mocking the weston_load_module() does return an object
+ *                      -# Calling the wet_module_init()
+ *                      -# Verification point:
+ *                         +# wet_module_init() must not return 0
+ *                         +# weston_plugin_api_get() must be called once time
+ *                         +# wl_global_create() must be called once time
+ *                         +# weston_load_module() must be called once time
+ */
+TEST_F(ControllerTests, wet_module_init_cannotInitIviInputModule)
+{
+    ControllerTests::ms_inputControllerModuleValue = -1;
+
+    struct ivi_layout_interface *lp_iviLayoutInterface = &g_iviLayoutInterfaceFake;
+    SET_RETURN_SEQ(weston_plugin_api_get, (const void**)&lp_iviLayoutInterface, 1);
+    struct wl_global *lp_wlGlobal = (struct wl_global*) 0xFFFFFFFF;
+    SET_RETURN_SEQ(wl_global_create, &lp_wlGlobal, 1);
+    void *lp_iviInputInitModule = (void*)custom_input_controller_module_init;
+    SET_RETURN_SEQ(weston_load_module, &lp_iviInputInitModule, 1);
+
+    ASSERT_NE(wet_module_init(&m_westonCompositor, nullptr, nullptr), 0);
+
+    ASSERT_EQ(weston_plugin_api_get_fake.call_count, 1);
+    ASSERT_EQ(wl_global_create_fake.call_count, 1);
+    // this API dont have anymore
+    ASSERT_EQ(weston_load_module_fake.call_count, 1);
```

Comment:
wl_array_release_fake.call_count should be verifed to check wet_module_init() failed at load_input_module() or not